Makefile

```diff
@@ -306,39 +308,15 @@ do-cover-file:
 
 test-te: do-cover-file
 	@echo Testing TE
-
-
 	@echo "Packages to test: "$(TE_PACKAGES)
-
-	@for package in $(TE_PACKAGES); do \
-		echo "Testing "$$package; \
-		$(GO) test $(GOFLAGS) -run=$(TESTS) $(TESTFLAGS) -test.v -test.timeout=2000s -covermode=count -coverprofile=cprofile.out -coverpkg=$(ALL_PACKAGES_COMMA) $$package || exit 1; \
-		if [ -f cprofile.out ]; then \
-			tail -n +2 cprofile.out >> cover.out; \
-			rm cprofile.out; \
-		fi; \
-	done
+	$(GO) test $(GOFLAGS) -run=$(TESTS) $(TESTFLAGS) -p 1 -v -timeout=2000s -covermode=count -coverpkg=$(ALL_PACKAGES_COMMA) -exec $(ROOT)/scripts/test-xprog.sh $(TE_PACKAGES)
```

Comment:
Without the -p flag, `go test` will run tests for different packages in parallel. This is actually really close to working! The other PR waiting on @crspeller has some of the packages testing against their own independent databases. Once that's merged and I update the other packages to do the same, I'll just need to make the server spin up with a random available listen port for the tests.